packages/eden/src/links/internal/operators.ts

```diff
@@ -36,60 +36,60 @@ export function share<TValue, TError>(
   _options?: ShareOptions,
 ): MonoTypeOperatorFunction<TValue, TError> {
   return (source) => {
-    let refCount = 0
+    let referenceCount = 0
 
     let subscription: Unsubscribable | null = null
 
     const observers: Partial<Observer<TValue, TError>>[] = []
 
     const startIfNeeded = () => {
-      if (subscription != null) return
+      if (subscription != undefined) return
 
       // Make shallow copy of observers in case they unsubscribe during the loop.
 
       subscription = source.subscribe({
         next: (value) => {
-          for (const observer of [...observers]) {
+          for (const observer of observers) {
```

Comment:
If you look on line 48, I mention that observers needs to be shallow cloned every time we need to iterate over them. This applies for all instances of iterating over them.